### PR TITLE
Allow MimeRenderDocuments to use base64 encoded files.

### PR DESCRIPTION
**Pull request by ian-r-rose**
This extends the `rendermime-interface` to be able to use other file formats than plain text (e.g., base64 encoded files).

My understanding of the `rendermime-interface` package is that it should have as few dependencies as possible, so I have typed `fileFormat` as a string rather than a `Contents.FileFormat`. If this is in error, happy to change it.

### CODE REVIEW

File: packages/rendermime-interfaces/src/index.ts [hunk 1]
```diff
@@ -83,6 +83,11 @@ namespace IRenderMime {
     readonly name: string;
 
     /**
+     * The name of the document model type.
+     */
+    readonly modelName: string;
```
Comment (blink1073) [inline packages/rendermime-interfaces/src/index.ts]: I think it is fair for both of these to be optional.

Comment (ian-r-rose) [inline packages/rendermime-interfaces/src/index.ts]: Sure. It was not clear to me exactly how closely these should match the interfaces given in the `DocumentRegistry`. They seem close but not exact.

Comment (blink1073) [inline packages/rendermime-interfaces/src/index.ts]: They are meant to be compatible, but not necessarily exact.

**Comment by blink1073** [comment]
Great, thanks!

**Pull request merged**